### PR TITLE
feat: Expiremental Warning Dialog

### LINKED ISSUE #2176: Add a dialog notifying that the app is prototype
Add popup screen: "This mobile app is still in its prototype stage with many features being experimental. We currently recommend using this app only for testing."

The popup should only display once

### PR DESCRIPTION
**Pull request by nileshteji**
Fixes #2176 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Screenshots for the change:
[Change Image](https://drive.google.com/file/d/1EDF1XW3jZX8X0-mIQypvIk91IcJskCzm/view?usp=sharing)

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 1]
```diff
@@ -46,24 +50,40 @@
...
+        sharedPreferences=requireActivity().getSharedPreferences(getResources().getString(R.string.warning_preference_key), Context.MODE_APPEND);
+        editor=sharedPreferences.edit();
         return binding.getRoot();
     }
 
     @Override
     public void onStart() {
         super.onStart();
 
+        if(!sharedPreferences.getBoolean(getResources().getString(R.string.warning_preference_key),false)){
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:77]: Codacy found an issue: ['if' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984&bid=20810369)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 2]
```diff
@@ -46,24 +50,40 @@
...
+            warning_alert_dialog=new AlertDialog.Builder(requireActivity())
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:78]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 3]
```diff
@@ -46,24 +50,40 @@
     private StartFragmentBinding binding;
     private Validator validator;
     private SharedViewModel sharedViewModel;
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences.Editor editor;
+    private AlertDialog warning_alert_dialog;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:55]: Codacy found an issue: [Name 'warning_alert_dialog' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 1 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:77]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 4]
```diff
@@ -46,24 +50,40 @@
...
     @Override
     public void onStart() {
         super.onStart();
 
+        if(!sharedPreferences.getBoolean(getResources().getString(R.string.warning_preference_key),false)){
+            warning_alert_dialog=new AlertDialog.Builder(requireActivity())
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message)
+                .setPositiveButton(R.string.ok,(dialog, which) -> {
+                    editor.putBoolean(getResources().getString(R.string.warning_preference_key),true);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:82]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 5]
```diff
@@ -46,24 +50,40 @@
...
     }
 
+    @SuppressLint("WrongConstant")
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = DataBindingUtil.inflate(inflater, R.layout.start_fragment, container, false);
         startFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(StartViewModel.class);
         sharedViewModel = ViewModelProviders.of(getActivity()).get(SharedViewModel.class);
         validator = new Validator(binding);
+        sharedPreferences=requireActivity().getSharedPreferences(getResources().getString(R.string.warning_preference_key), Context.MODE_APPEND);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:68]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 3 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:55]: Codacy found an issue: [Avoid unused private fields such as 'warning_alert_dialog'.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 6]
```diff
@@ -46,24 +50,40 @@
...
+        editor=sharedPreferences.edit();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:69]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 7]
```diff
@@ -46,24 +50,40 @@
...
 
     @Override
     public void onStart() {
         super.onStart();
 
+        if(!sharedPreferences.getBoolean(getResources().getString(R.string.warning_preference_key),false)){
+            warning_alert_dialog=new AlertDialog.Builder(requireActivity())
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message)
+                .setPositiveButton(R.string.ok,(dialog, which) -> {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:81]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984&bid=20810369)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 5 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:68]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 4 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:82]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 6 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:69]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 3 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:55]: Codacy found an issue: [Name 'warning_alert_dialog' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 7 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:81]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 1 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:77]: Codacy found an issue: ['if' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:77]: Codacy found an issue: [',' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 3 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:55]: Codacy found an issue: [Avoid unused private fields such as 'warning_alert_dialog'.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 2 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:78]: Codacy found an issue: ['=' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984&bid=20810369)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 8]
```diff
@@ -46,24 +50,39 @@
...
+        sharedPreferences = requireActivity().getSharedPreferences(getResources().getString(R.string.warning_preference_key), Context.MODE_APPEND);
+        editor = sharedPreferences.edit();
         return binding.getRoot();
     }
 
     @Override
     public void onStart() {
         super.onStart();
-
+        if (!sharedPreferences.getBoolean(getResources().getString(R.string.warning_preference_key), false)){
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:76]: Codacy found an issue: ['if' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 9]
```diff
@@ -46,24 +50,39 @@
     private StartFragmentBinding binding;
     private Validator validator;
     private SharedViewModel sharedViewModel;
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences.Editor editor;
+    AlertDialog warningAlertDialog;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:55]: Codacy found an issue: [Variable 'warningAlertDialog' must be private and have accessor methods.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=6423984)

**Comment by nileshteji** [comment]
@iamareebjamal Sir can you please  review the PR.Thank you

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 10]
```diff
@@ -46,24 +50,39 @@
...
     private SharedViewModel sharedViewModel;
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences.Editor editor;
+
 
     public static StartFragment newInstance() {
         return new StartFragment();
     }
 
+    @SuppressLint("WrongConstant")
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Remove

File: app/src/main/res/values-zh-rCN/strings.xml [hunk 11]
```diff
@@ -421,6 +421,8 @@
     <string name="third_party_licenses">使用的第三方库</string>
     <string name="event_owner_detail_organiser">组织者</string>
     <string name="past_event_error_message">Past events cannot be published.</string>
+    <string name="warning_preference_key">experimental_warning_shown</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values-zh-rCN/strings.xml]: You don't store keys in strings.xml

File: app/src/main/res/values-zh-rCN/strings.xml [hunk 12]
```diff
@@ -421,6 +421,8 @@
...
+    <string name="dialog_message">This mobile app is still in its prototype stage with many features being experimental. We currently recommend using this app only for testing.</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values-zh-rCN/strings.xml]: This is not chinese, why is it in Chinese strings?

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -421,6 +421,8 @@
     <string name="third_party_licenses">Third Party Libraries Used</string>
     <string name="event_owner_detail_organiser">Organiser</string>
     <string name="past_event_error_message">Past events cannot be published.</string>
+    <string name="warning_preference_key">experimental_warning_shown</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: You don't store keys in strings.xml

File: app/src/main/res/values-zh-rCN/strings.xml (hunk 12 above)
Comment (nileshteji) [reply app/src/main/res/values-zh-rCN/strings.xml]: @iamareebjamal Sir if I am not storing the message string in Chinese Strings File .It is failing the Build.

File: app/src/main/res/values/strings.xml (hunk 13 above)
Comment (nileshteji) [reply app/src/main/res/values/strings.xml]: okay sir

File: app/src/main/res/values-zh-rCN/strings.xml (hunk 12 above)
Comment (nileshteji) [reply app/src/main/res/values-zh-rCN/strings.xml]: what I can do is translate the string to Chinese and then add to the Chinese Strings.

Comment (iamareebjamal) [reply app/src/main/res/values-zh-rCN/strings.xml]: Disable that lint

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 14]
```diff
@@ -46,24 +49,40 @@
...
 
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = DataBindingUtil.inflate(inflater, R.layout.start_fragment, container, false);
         startFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(StartViewModel.class);
         sharedViewModel = ViewModelProviders.of(getActivity()).get(SharedViewModel.class);
         validator = new Validator(binding);
+        sharedPreferences = requireActivity().getSharedPreferences(warningPreferenceKey, Context.MODE_PRIVATE);
+        editor = sharedPreferences.edit();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Write this logic in ViewModel

Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java:69]: Sir having the logic in viewModel. We have to deal with Shared Preferences as a live data which I think is of no use.
Thank you

Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Okay Sir

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 15]
```diff
@@ -63,7 +66,17 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void onStart() {
         super.onStart();
-
+        startFragmentViewModel.getWarningDialog().observe(this, prefernceKey -> {
+            if (!prefernceKey) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: No logic should be placed here. Simply send the callback only if preference key is not set

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 16]
```diff
@@ -37,6 +37,8 @@
     ;
     private final SingleEventLiveData<Boolean> isEmailRegistered = new SingleEventLiveData<>();
     private final MutableLiveData<EmailRequest> emailRequestModel = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> warningDialogLiveData=new MutableLiveData<>();
+    private String warningPreferenceKey = "experimental_warning_shown";
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: static final

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: And capitalize

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 17]
```diff
@@ -37,6 +37,8 @@
     ;
     private final SingleEventLiveData<Boolean> isEmailRegistered = new SingleEventLiveData<>();
     private final MutableLiveData<EmailRequest> emailRequestModel = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> warningDialogLiveData=new MutableLiveData<>();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Follow proper style guidelines

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 18]
```diff
@@ -102,6 +104,15 @@ public void setBaseUrl(String url, boolean shouldSetDefaultUrl) {
         sharedPreferenceModel.saveString(Constants.SHARED_PREFS_BASE_URL, baseUrl);
     }
 
+    public MutableLiveData<Boolean> getWarningDialog() {
+        warningDialogLiveData.setValue(sharedPreferenceModel.getBoolean(warningPreferenceKey,false));
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Didn't add to disposable

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 16 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: okay

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 15 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Okay Sir

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 17 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Okay

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 14 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: sir should handle the AlertDialog in a method or it is okay to call them in the lambda itself

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: lambda

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 18 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: We are not dealing Observables so adding to disposable will not be required.Thank you

Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: We are dealing with same type of scenario in isLoggedIn LiveData.

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 19]
```diff
@@ -63,7 +66,16 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void onStart() {
         super.onStart();
+        startFragmentViewModel.getWarningDialog().observe(this, prefernceKey -> {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]:
```suggestion
        startFragmentViewModel.getWarningDialog().observe(this, showMessage -> {
```

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 20]
```diff
@@ -37,6 +37,8 @@
     ;
     private final SingleEventLiveData<Boolean> isEmailRegistered = new SingleEventLiveData<>();
     private final MutableLiveData<EmailRequest> emailRequestModel = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> showWarningDialog=new MutableLiveData<>();
+    private static final String warningPreferenceKey = "experimental_warning_shown";
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Capitalize

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 21]
```diff
@@ -37,6 +37,8 @@
     ;
     private final SingleEventLiveData<Boolean> isEmailRegistered = new SingleEventLiveData<>();
     private final MutableLiveData<EmailRequest> emailRequestModel = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> showWarningDialog=new MutableLiveData<>();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Proper style

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 20 above)
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]:
```suggestion
    private static final String warningPreferenceKey = "show_experimental_warning";
```

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java (hunk 19 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Okay

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 20 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Capitalizing the String will not adhere to the coding style of the CodeBase . As it is following Camel Case in the Whole Project.Thank You

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 21 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: In this Should the name of the live data be changed or the casing of the variable?

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 20 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: Or Writing **WARNINGPREFERENCEKEY** would be accepted?

Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: okay done

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java (hunk 21 above)
Comment (nileshteji) [reply app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]: @iamareebjamal  can you have a look at it

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 22]
```diff
@@ -63,7 +66,16 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void onStart() {
         super.onStart();
+        startFragmentViewModel.getWarningDialog().observe(this, showmessage -> {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]:
```suggestion
        startFragmentViewModel.getWarningDialog().observe(this, showMessage -> {
```

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java [hunk 23]
```diff
@@ -63,7 +66,16 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+             new AlertDialog.Builder(requireActivity())
+                    .setTitle("Message")
+                    .setMessage(R.string.dialog_message)
+                    .setPositiveButton(R.string.ok, ((dialog, which) -> {
+                        startFragmentViewModel.setWarningPreference();
+                        dialog.dismiss();
+                    })).show();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java]: Move this block to line 90

File: app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java [hunk 24]
```diff
@@ -37,6 +37,8 @@
     ;
     private final SingleEventLiveData<Boolean> isEmailRegistered = new SingleEventLiveData<>();
     private final MutableLiveData<EmailRequest> emailRequestModel = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> warningDialog=new MutableLiveData<>();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/start/StartViewModel.java]:
```suggestion
    private final MutableLiveData<Boolean> warningDialog = new MutableLiveData<>();
```

**Review by iamareebjamal — DISMISSED** [review]

**Comment by nileshteji** [comment]
@iamareebjamal Sorry Sir I closed this pull request by mistake .

**Pull request merged**